pkg/controller/podautoscaler/replica_calculator_test.go

```diff
@@ -425,6 +462,36 @@ func TestReplicaCalcScaleUpCMObject(t *testing.T) {
 	tc.runTest(t)
 }
 
+func TestReplicaCalcScaleUpCMExternal(t *testing.T) {
+	tc := replicaCalcTestCase{
+		currentReplicas:  1,
+		expectedReplicas: 2,
+		metric: &metricInfo{
+			name:                "qps",
+			levels:              []int64{8600},
+			targetUtilization:   4400,
+			expectedUtilization: 8600,
+			selector:            &metav1.LabelSelector{},
+		},
+	}
+	tc.runTest(t)
+}
+
+func TestReplicaCalcScaleUpPerPodCMExternal(t *testing.T) {
+	tc := replicaCalcTestCase{
+		currentReplicas:  3,
+		expectedReplicas: 4,
+		metric: &metricInfo{
+			name:                    "qps",
+			levels:                  []int64{8600},
+			perPodTargetUtilization: 2150,
+			expectedUtilization:     2867,
+			selector:                &metav1.LabelSelector{},
```

Comment:
can we get a test that uses LabelSelector and verifies that it's passed through correctly?